connect/api/src/main/java/org/apache/kafka/connect/data/SchemaBuilder.java

```diff
@@ -185,7 +184,7 @@ public SchemaBuilder doc(String doc) {
 
     @Override
     public Map<String, String> parameters() {
-        return parameters == null ? null : Collections.unmodifiableMap(parameters);
+        return parameters == null ? null : Map.copyOf(parameters);
```

Comment:
According to comment in line#198, the order is kept as expected. Perhaps, we should keep using `Collections.unmodifiableMap`